modules/apps/commerce/commerce-order-content-web/test/components/AddDeliveryGroupButton/AddDeliveryGroupButton.spec.tsx

```diff
@@ -0,0 +1,192 @@
+			(): IPostalAddressAPIResponse => {
+				return {
+					items: [
+						{
+							addressCountry: 'United States',
+							addressLocality: 'addressLocality1',
+							addressRegion: 'Alabama',
+							addressType: 'billing-and-shipping',
+							externalReferenceCode:
+								'71061669-ba97-943c-70a3-96cdc0c8305a',
+							id: 101,
+							name: 'name1',
+							phoneNumber: 'phoneNumber1',
+							postalCode: 'postalCode1',
+							primary: false,
+							streetAddressLine1: 'streetAddressLine11',
+							streetAddressLine2: 'streetAddressLine21',
+							streetAddressLine3: 'streetAddressLine31',
+						},
+					],
+				};
+			}
+		);
+	});
+
+	afterEach(() => {
+		fetchMock.restore();
+		jest.clearAllMocks();
+
+		cleanup();
+	});
+
+	it('Must check modal opening', async () => {
+		const renderedComponent = render(
+			<AddDeliveryGroupButton
+				accountId={10}
+				handleSubmit={handleSubmit}
+			/>
+		);
+
+		const addDeliveryGroupButton = renderedComponent.getByRole('button', {
+			name: 'add-delivery-group',
+		});
+
+		await act(async () => {
+			addDeliveryGroupButton.click();
+		});
+
+		await waitFor(() => {
+			expect(
+				renderedComponent.getByRole('button', {name: 'save'})
+			).toBeVisible();
+		});
+
+		const {saveButton} = deliveryGroupModalLocateFields(renderedComponent);
+
+		expect(saveButton).toBeVisible();
+	});
+
+	it('Must close modal on cancel', async () => {
+		const renderedComponent = render(
+			<AddDeliveryGroupButton
+				accountId={10}
+				handleSubmit={handleSubmit}
+			/>
+		);
+
+		const addDeliveryGroupButton = renderedComponent.getByRole('button', {
+			name: 'add-delivery-group',
+		});
+
+		await act(async () => {
+			addDeliveryGroupButton.click();
+		});
+
+		await waitFor(() => {
+			expect(
+				renderedComponent.getByRole('button', {name: 'cancel'})
+			).toBeVisible();
+		});
+
+		const {cancelButton} =
+			deliveryGroupModalLocateFields(renderedComponent);
+
+		await act(async () => {
+			cancelButton.click();
+		});
+
+		await waitFor(() => {
+			expect(screen.queryByText('cancel')).not.toBeInTheDocument();
+		});
+	});
+
+	it('Must close modal on submit', async () => {
+		const renderedComponent = render(
+			<AddDeliveryGroupButton
+				accountId={10}
+				handleSubmit={handleSubmit}
+			/>
+		);
+
+		const addDeliveryGroupButton = renderedComponent.getByRole('button', {
+			name: 'add-delivery-group',
+		});
+
+		await act(async () => {
+			addDeliveryGroupButton.click();
+		});
+
+		await waitFor(() => {
+			expect(
+				renderedComponent.getByRole('button', {name: 'save'})
+			).toBeVisible();
+		});
+
+		const {addressIdSelect, deliveryGroupNameInput, saveButton} =
+			deliveryGroupModalLocateFields(renderedComponent);
+
+		await setFieldValue(addressIdSelect, String(101));
+		await setFieldValue(deliveryGroupNameInput, 'deliveryGroupName');
+
+		await act(async () => {
+			saveButton.click();
+		});
+
+		expect(handleSubmit).toBeCalledTimes(1);
```

Comment:
This might be vacuously a `pass`. Why not include in `waitFor`?

Reply:
🤔